extensions/ql-vscode/src/codeql-cli/cli.ts

```diff
@@ -159,6 +159,24 @@ export type ResolvedQueries = string[];
  */
 export type ResolvedTests = string[];
 
+/**
+ * A compilation message for a test message (either an error or a warning)
+ */
+export interface CompilationMessage {
```

Comment:
Should this go in `new-messages.ts`?

Also should we rename `new-messages.ts` (and I guess `new-query-runner.ts`)? Happy for that to be a follow up.

Reply:
Not really, since this is not a message that is produced by the query server. This message is only produced for tests, and never as part of a query run. Therefore, I don't think it makes sense to move it to `new-messages.ts`.

I can rename `new-messages.ts` in a follow-up PR. `new-query-runner.ts` has already been renamed as part of https://github.com/github/vscode-codeql/pull/3171